src/robot/libraries/OperatingSystem.py

```diff
@@ -286,34 +287,36 @@ def grep_file(self, path, pattern, encoding='UTF-8', encoding_errors='strict'):
         Possible trailing newline is never returned.
 
         A line matches if it contains the ``pattern`` anywhere in it and
-        it *does not need to match the pattern fully*. The pattern
-        matching syntax is explained in `introduction`, and in this
+        it *does not need to match the pattern fully when not in regex mode*. 
+        The pattern matching syntax is explained in `introduction`, and in this
         case matching is case-sensitive.
```

Comment:
1. Shouldn't partial matching work also in the regexp mode? Pattern type affecting this would be somewhat inconsistent and AFAIK `grep` on the command line supports partial matching as well.
2. A bit more information is needed about regexp matching here. Should probably explain both patterns separately so that glob explanation refers to the docs in the introduction and regexp is briefly explained here. I can fine tune this later if needed as well.
3. A note either here or at the end of the documentation explaining that regexp support is new in RF 5.0 is needed.

Reply:
Did some updates to the documentation, should be clearer now